Libraries/Network/NetInfo.js

```diff
@@ -233,14 +230,25 @@ const NetInfo = {
     },
   },
 
-  isConnectionExpensive(callback: (metered: ?boolean, error?: string) => void): void {
+  isConnectionExpensive(): Promise {
+    if (arguments.length > 1) {
+      console.warn('NetInfo.isConnectionMetered(callback) is deprecated. Use the returned Promise instead.');
+      const callback = arguments[1];
+      if (Platform.OS === 'android') {
+        RCTNetInfo.isConnectionMetered().then(
+          metered => callback(metered),
+          error => callback(null, error)
+        );
+      } else {
+        callback(null, 'Unsupported');
+      }
+      return;
```

Comment:
> temporary things tend to get more permanent than we plan

Yeah, not sure how I can tackle this, maybe I need to keep a tasklist like we do for commits to cherry-pick, and ensure that they are removed.

> we will do this if (callbacks.length > 0) a lot, for every API that we want to promisify. If we don't provide a helper function we will have much more almost dead code in APIs for quite a while.

Yeah, we're going to do this a lot, so the helper method can indeed come handy. But given the inconsistency between the types of callbacks, it won't be that much helpful IMO - https://github.com/facebook/react-native/issues/4971